src/Quoter.sol

```diff
@@ -13,6 +16,9 @@ import { PeripheryImmutableState } from "src/abstract/PeripheryImmutableState.so
 
 contract Quoter is IQuoter, PeripheryImmutableState
 {
+    using FactoryStoreLib for GenericFactory;
```

Comment:
Can't do FactoryStoreLib for IGenericFactory?

Reply:
the issue is, the `read` function in `FactoryStoreLib` has `GenericFactory` as the argument type. So if we want to use `IGenericFactory` we have to: 

- make a copy of that lib locally and adapt that for `IGenericFactory`. But that will be duplicating and repeating ourselves by storing the same functionality in both `v3-core` and `v3-periphery`

Reply:
And you've tried just making it `IGenericFactory` everywhere? I.e. can you pass GenericFactory into a function that expect IGenericFactory

Reply:
yup just tried. I thought of this but avoided it cuz I wanted to minimize periphery changing core contracts. 

It works on both periphery and core contracts with no issues. 

Should be all good now.